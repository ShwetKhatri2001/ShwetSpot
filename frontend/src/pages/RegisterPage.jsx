import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { Button, Col, Form, Row } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';
import Message from '../components/Message';
import Loader from '../components/Loader';
import { getUserDetails, register } from '../redux/actions/userActions';

const RegisterPage = ({ location, history }) => {
  const [name, setName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');
  const [message, setMessage] = useState(null);

  const dispatch = useDispatch();

  const userRegister = useSelector(state => state.userRegister);
  const { loading, error, userInfo } = userRegister;

  const redirect = location.search ? location.search.split('=')[1] : '/';

  useEffect(() => {
    if (userInfo) {
      history.push(redirect);
      dispatch(getUserDetails('profile'));
    }
  }, [history, userInfo, redirect, dispatch]);

  const submitHandler = e => {
    e.preventDefault();
    if (password !== confirmPassword) {
      setMessage('Passwords do not match');
    } else {
      dispatch(register(name, email, password));
    }
  };

  return (
    <div className="justify-content-center align-items-center container bg-light p-4">
      <h1>Sign Up</h1>
      {message && <Message variant="danger">{message}</Message>}
      {error && <Message variant="danger">{error}</Message>}
      {loading && <Loader />}
      <Form onSubmit={submitHandler}>
        <Form.Group controlId="name">
          <Form.Label>Name</Form.Label>
          <Form.Control
            type="name"
            placeholder="Enter name"
            value={name}
            onChange={e => setName(e.target.value)}
          />
        </Form.Group>

        <Form.Group controlId="email">
          <Form.Label>Email Address</Form.Label>
          <Form.Control
            type="email"
            placeholder="Enter email"
            value={email}
            onChange={e => setEmail(e.target.value)}
          />
        </Form.Group>

        <Form.Group controlId="password">
          <Form.Label>Password</Form.Label>
          <Form.Control
            type="password"
            placeholder="Enter password"
            value={password}
            onChange={e => setPassword(e.target.value)}
          />
        </Form.Group>

        <Form.Group controlId="confirmPassword">
          <Form.Label>Confirm Password</Form.Label>
          <Form.Control
            type="password"
            placeholder="Confirm password"
            value={confirmPassword}
            onChange={e => setConfirmPassword(e.target.value)}
          />
        </Form.Group>

        <Button type="submit" variant="success">
          Register
        </Button>
      </Form>

      <Row className="py-3">
        <Col>
          Already have an account?{' '}
          <Link to={redirect ? `/login?redirect=${redirect}` : '/login'}>
            Log in
          </Link>
        </Col>
      </Row>
    </div>
  );
};

export default RegisterPage;
