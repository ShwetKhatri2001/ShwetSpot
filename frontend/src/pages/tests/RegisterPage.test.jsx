import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import RegisterPage from '../RegisterPage';
import store from '../../redux/store';

describe('RegisterPage Component', () => {
  const MockRegisterPage = (
    <MemoryRouter>
      <Provider store={store}>
        <RegisterPage location={{ search: '1' }} />
      </Provider>
    </MemoryRouter>
  );

  it('should match the snapshot', () => {
    const tree = renderer.create(MockRegisterPage).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
