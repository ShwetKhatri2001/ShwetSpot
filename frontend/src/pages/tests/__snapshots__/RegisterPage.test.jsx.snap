// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterPage Component should match the snapshot 1`] = `
<div
  className="justify-content-center align-items-center container bg-light p-4"
>
  <h1>
    Sign Up
  </h1>
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="name"
      >
        Name
      </label>
      <input
        className="form-control"
        id="name"
        onChange={[Function]}
        placeholder="Enter name"
        type="name"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="email"
      >
        Email Address
      </label>
      <input
        className="form-control"
        id="email"
        onChange={[Function]}
        placeholder="Enter email"
        type="email"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form-control"
        id="password"
        onChange={[Function]}
        placeholder="Enter password"
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
        htmlFor="confirmPassword"
      >
        Confirm Password
      </label>
      <input
        className="form-control"
        id="confirmPassword"
        onChange={[Function]}
        placeholder="Confirm password"
        type="password"
        value=""
      />
    </div>
    <button
      className="btn btn-success"
      disabled={false}
      type="submit"
    >
      Register
    </button>
  </form>
  <div
    className="py-3 row"
  >
    <div
      className="col"
    >
      Already have an account?
       
      <a
        href="/login"
        onClick={[Function]}
      >
        Log in
      </a>
    </div>
  </div>
</div>
`;
